src/graphtypes/simplegraphs/simpledigraph.jl

```diff
@@ -118,9 +118,9 @@ end
 
 function rem_edge!(g::SimpleDiGraph, e::SimpleDiGraphEdge)
     has_edge(g, e) || return false
-    i = searchsorted(g.fadjlist[src(e)], dst(e))[1]
+    i = searchsortedfirst(g.fadjlist[src(e)], dst(e))
```

Comment:
i think this does twice the work. compare https://github.com/mschauer/CausalInference.jl/blob/master/src/skeleton.jl#L11

also https://github.com/JuliaGraphs/LightGraphs.jl/blob/f435476d0d8b72ceec712e2876370c5503568e3d/src/graphtypes/simplegraphs/simplegraph.jl#L143 could check for vertex degree before searching as in has_edge

Reply:
It *does* do twice the work - nice catch. This was a remnant from when we had edge sets and `has_edge` was O(1). Let me fix that.

Reply:
i like `i = searchsorted(); isempty(i) && return false; j = first(i)` more, but it should be equivalent.